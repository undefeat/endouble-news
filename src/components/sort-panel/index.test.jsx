import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import shallow from 'react-test-renderer/shallow';
import SortPanel from '.';

// eslint-disable-next-line
class SortPanelWrapped extends React.Component {
    // eslint-disable-next-line
    render() {
        return <SortPanel {...this.props} />;
    }
}

describe('SortPanel', () => {
    it('should correctly render', () => {
        const tree = shallow
            .createRenderer()
            .render(<SortPanel sortBy="date" sortOptions={['date', 'source']} onSortByChanged={() => {}} />);

        expect(tree).toMatchSnapshot();
    });

    it('should call "onSortByChanged" when click', () => {
        const onSortByChanged = jest.fn();
        const element = ReactTestUtils.renderIntoDocument(
            // ReactTestUtils don't render functional components.
            <SortPanelWrapped sortBy="date" sortOptions={['date', 'source']} onSortByChanged={onSortByChanged} />,
        );

        const btn = ReactTestUtils.findRenderedDOMComponentWithClass(element, 'sort-panel__btn');

        ReactTestUtils.Simulate.click(btn);

        expect(onSortByChanged).toHaveBeenCalledTimes(1);
        expect(onSortByChanged).toHaveBeenCalledWith('source');
    });
});
