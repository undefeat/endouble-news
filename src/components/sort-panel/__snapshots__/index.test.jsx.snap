// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortPanel should correctly render 1`] = `
<div
  className="sort-panel"
>
  <button
    className="sort-panel__btn"
    onClick={[Function]}
    type="button"
  >
    <span>
      Sort by 
    </span>
    <span>
      source
    </span>
  </button>
</div>
`;
